simple_registers/counters/counter16_4/test_counter16_4.py

```diff
@@ -0,0 +1,237 @@
+#####################################################################
+# Test the functionality of 16 bit Counter
+#####################################################################
+import random
+import numpy as np
+import cocotb
+from cocotb.binary import BinaryValue
+from cocotb.clock import Clock
+from cocotb.triggers import Timer, ClockCycles
+from cocotb.triggers import RisingEdge
+from cocotb.triggers import FallingEdge
+
+@cocotb.test()
+async def test_counter16_4(dut):
+
+	################################################################
+	# Clock Generation
+	CLK0_PERIOD = 10 # [ns]
+	cocotb.start_soon(Clock(dut.clock0, CLK0_PERIOD, units="ns").start())
+	CLK1_PERIOD = 20 # [ns]
+	cocotb.start_soon(Clock(dut.clock1, CLK1_PERIOD, units="ns").start())
+	CLK2_PERIOD = 30 # [ns]
+	cocotb.start_soon(Clock(dut.clock2, CLK2_PERIOD, units="ns").start())
+	CLK3_PERIOD = 40 # [ns]
+	cocotb.start_soon(Clock(dut.clock3, CLK3_PERIOD, units="ns").start())
+
+	################################################################
+	
+	# Counter0 test
+	################################################################
+	test_cases = 2
```

Comment:
See if you can create a python function for testing counters. Then you call this function frequently. Just like what I did for forcing bitstreams.

Not required in this PR. Consider to try in later PRs.